image/store.go

```diff
@@ -247,7 +269,9 @@ func (is *store) Delete(id ID) ([]layer.Metadata, error) {
 	if err := is.digestSet.Remove(id.Digest()); err != nil {
 		logrus.Errorf("error removing %s from digest set: %q", id, err)
 	}
+
 	delete(is.images, id)
+	delete(is.imageCache, id)
```

Comment:
In addition to this deletion on `Delete()`, the cache should be also periodically deleted?

Reply:
We might be able to use https://github.com/hashicorp/golang-lru

Reply:
I've been considering using this lib days before, but I'm not sure whether it worth doing considering the complexity we introduce. Anyway, there are still bug in the current implementation, I'm still working on it and I'll do more test later.

Reply:
@AkihiroSuda I just replace map with fixed size golang-lru cache, the cache size if set to 1024.